src/preamble.js

```diff
@@ -2303,6 +2313,10 @@ function integrateWasmJS() {
       assert(Module === trueModule, 'the Module object should not be replaced during async compilation - perhaps the order of HTML elements is wrong?');
       trueModule = null;
 #endif
+#if USE_PTHREADS
+      // Keeep a reference to the compiled module so we can post it to the workers.
+      Module['wasmModule'] = output['module'];
```

Comment:
the module should be saved as `Module['asm']` already

Reply:
The file `preamble.js` has `Module['asm'] = exports;`, so `'asm'` seems to be getting the exports object, rather than the Module. So it looks like we do need to store this manually(?)

I think what we'd like is to make `receiveInstance()` take a second parameter that is the module, so that applications that instantiate themselves will also get the module stored for threads.

I do hate this a bit that we do have to store the module to be able to instantiate a new thread - that wastes memory; and I was pondering if we could reuse IndexedDB and the thread pool idea to optimize, i.e. if application knows they only need fixed K threads at startup, the could precreate a pool of that size to be able to throw away the Module afterwards. If new threads want to spawn after that, they would take a slow path of reading the Module from IndexedDB for instantiation. Although not sure if that's worth the trouble to optimize, so let's leave that for later.